### PR TITLE
Models and DB were added

### PR DESCRIPTION
**Pull request by Kotsiuba**

### CODE REVIEW

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Tag_Item>().HasKey(t => new { t.ItemId, t.TagId});
+        }
+
+        public DbSet<Item> Items { get; set; }
+        public DbSet<ItemCategory> Categories { get; set; }
+        public DbSet<Tag> Tags { get; set; }
+        public DbSet<Tag_Item> Tags_Items { get; set; }
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs]: EF can implement many-to-many without implicit setup. Remove it.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public class ToDoDbContext : DbContext
+    {
+        public ToDoDbContext(DbContextOptions dbContextOptions) : base(dbContextOptions)
+        {
+
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Tag_Item>().HasKey(t => new { t.ItemId, t.TagId});
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs]: Remove it.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+using ToDo.DataAccess.Models;
+
+namespace ToDo.DataAccess.DataBase
+{
+    public class ToDoDbInitializer
+    {
+        public static void Initialize(ToDoDbContext context)//SchoolContext is EF context
+        {
+
+            context.Database.EnsureCreated();//if db is not exist ,it will create database .but ,do nothing .
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs]: Remove comments from code.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+using System.Data.Entity;
+using System.Linq;
+using System.Threading.Tasks;
+using ToDo.DataAccess.Models;
+
+namespace ToDo.DataAccess.DataBase
+{
+    public class ToDoDbInitializer
+    {
+        public static void Initialize(ToDoDbContext context)//SchoolContext is EF context
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs]: Remove comments from code.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+            var con1 = new Tag_Item { Item = item1, Tag = tag1 };
+            var con2 = new Tag_Item { Item = item1, Tag = tag2 };
+            var con3 = new Tag_Item { Item = item2, Tag = tag1 };
+            var con4= new Tag_Item { Item = item2, Tag = tag2 };
+
+            context.Items.AddRange(item1, item2, item3, item3);
+            context.Categories.AddRange(cat1, cat2, cat3);
+            context.Items.AddRange(item1, item2, item3, item3);
+            context.Items.AddRange(item1, item2, item3, item3);
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs]: Why do you have this line doubled?

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        SuperHigh,
+        High,
+        Medium,
+        Low,
+        SuperLow        
+    }
+
+    public class Item
+    {
+        public int Id { get; set; }
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Use `Guid` instead. For every ID.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        public string Title { get; set; }
+        public  string Description { get; set; }
+        public ItemPriority Priority { get; set; }
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Remove `Item` from name.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        public DateTime DueDate { get; set; }
+        public ItemCategory Caregory { get; set; }
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Remove `Item` from name.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        public string Status { get; set; }
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Status need to be enum or bool.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Tag.cs [hunk 10]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+
+namespace ToDo.DataAccess.Models
+{
+    public class Tag
+    {
+        public int Id{ get; set; }
+        public string Name { get; set; }
+        public string Color{ get; set; }
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/Models/Tag.cs]: fix spaces formatting
`public type Name { get; set; }`

File: ToDo.DataAccess/ToDo.DataAccess/Models/Tag_Item.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ToDo.DataAccess.Models
+{
+    public class Tag_Item
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/Models/Tag_Item.cs]: Remove at all.

File: ToDo.DataAccess/ToDo.DataAccess/Pages/About.cshtml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+﻿@page
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/Pages/About.cshtml]: Remove this project at all. I asked only one project with DataAccess. This is only EF and nothing more.

**Review by zhikin2207 — CHANGES_REQUESTED** [review]
This is absolutely different from what I wanted. Please take a look at the initial task and review comments.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs (hunk 1 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs]: This is not yet possible in EF Core without using an explicit class for the join. See here for an example of how to do that.


https://stackoverflow.com/questions/46184678/fluent-api-many-to-many-in-entity-framework-core-2-0

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs (hunk 2 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs]: It is needed for connection.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs (hunk 3 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs]: Forgot about it, deleted.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs (hunk 4 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs]: Deleted.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs (hunk 5 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbInitializer.cs]: Just excess ctrl+v, deleted.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs (hunk 8 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Your variant looks better, renamed.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs (hunk 9 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Changed to bool.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Tag.cs (hunk 10 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/Models/Tag.cs]: Fixed.

File: ToDo.DataAccess/ToDo.DataAccess/Pages/About.cshtml (hunk 12 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/Pages/About.cshtml]: The new project was created.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs [hunk 13]
```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.EntityFrameworkCore;
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs:1]: Remove unused usings everywhere.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Tag.cs (hunk 10 above)
Comment (zhikin2207) [reply ToDo.DataAccess/ToDo.DataAccess/Models/Tag.cs]: image

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs [hunk 14]
```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ToDo.DataAccess.DataBase;
+using ToDo.DataAccess.Models;
+
+namespace ToDo.DataAccess
+{
+    public class ToDoRepository : IToDoRepository
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: The repository should be created for every table or group of tables. Not for all application. So, please, create it for categories, items and tags.

Comment (zhikin2207) [reply ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: But in this case I would recommend to create `GenericRepository` for CRUD operations.

**Review by zhikin2207 — CHANGES_REQUESTED** [review]

File: ToDo.DataAccess/ToDo.DataAccess/Models/Tag.cs (hunk 10 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/Models/Tag.cs]: Surely fixed.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs (hunk 7 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Renamed.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs (hunk 13 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs:1]: this namespace is needed for db, others unused namespaces were removed

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs (hunk 6 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Changed.

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs (hunk 14 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: Changed.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+    public class ToDoDbContext : DbContext
+    {
+        public ToDoDbContext(DbContextOptions dbContextOptions) : base(dbContextOptions)
+        {
+
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<TagItem>().HasKey(t => new { t.ItemId, t.TagId});
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs]: Space around `{}`

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs [hunk 16]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace ToDo.DataAccess.Models
+{
+    public enum Priority
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Move into separate file.

File: ToDo.DataAccess/ToDo.DataAccess/Models/TagItem.cs [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace ToDo.DataAccess.Models
+{
+    public class TagItem
+    {
+        public Guid Id { get; set; }
+        public int ItemId { get; set; }
+        public Item Item { get; set; }
+
+        public int TagId { get; set; }
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/Models/TagItem.cs]: Change all ID to `Guid`

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs [hunk 18]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ToDo.DataAccess.DataBase;
+using ToDo.DataAccess.Models;
+
+namespace ToDo.DataAccess
+{
+    public class ToDoRepository : IDataRepository
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: Rename repository to `GenericRepository<T>`

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        private readonly ToDoDbContext _context;
+
+        ToDoRepository(ToDoDbContext newContext)
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: Make constructor public.

Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: Rename parameter `newContext` to `context`

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            _context = newContext;
+        }
+
+        public void Add<TEntity>(TEntity entity) where TEntity : class
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: Move all `T` constraints to class level

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+            _context.Set<TEntity>(); // TODO
+        }
+
+        public IEnumerable<TEntity> GetAll<TEntity>() where TEntity : class
+        {
+            return _context.Set<TEntity>(); 
+        }
+    }
+
+    public interface IDataRepository
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: Move into a separate file.

**Review by zhikin2207 — CHANGES_REQUESTED** [review]

File: ToDo.DataAccess/ToDo.DataAccess/Models/TagItem.cs (hunk 17 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/Models/TagItem.cs]: Changed.

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs (hunk 19 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: changed

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs (hunk 18 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: Renamed.

File: ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs (hunk 16 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/Models/Item.cs]: Moved.

File: ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs (hunk 15 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/DataBase/ToDoDbContext.cs]: Removed.

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs (hunk 19 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: renamed.

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs (hunk 20 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: Moved.

File: ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs (hunk 21 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/ToDoRepository.cs]: Moved.

**Review by zhikin2207 — APPROVED** [review]

**Pull request merged**